src/Analyzer.Utilities.UnitTests/JsonLineNumberResolverTests.cs

```diff
@@ -342,5 +342,293 @@ public void Constructor_NullTemplateContext_ThrowsException()
+                new TemplateContext
+                {
+                    OriginalTemplate = templateContext.OriginalTemplate,
+                    ExpandedTemplate = templateContext.ExpandedTemplate,
+                    IsMainTemplate = false,
+                    ParentContext = null
+                })
+                .ResolveLineNumber("path");
+        }
+
+        [DataTestMethod]
+        [DataRow("resources", 0, new object[] { "resources" }, DisplayName = "Parent template resources property found")]
+        [DataRow("resources[0].properties.template", 0, new object[] { "resources", 0, "properties", "template"}, DisplayName = "Parent template resource property found")]
+        [DataRow("resources[0].properties.siteConfig.ftpsState", 1, new object[] { "resources", 0, "properties", "template", "resources", 0, "properties", "siteConfig", "ftpsState" }, DisplayName = "One level nested template resource property found")]
+        [DataRow("parameters.ftpsState.defaultValue", 1, new object[] { "resources", 0, "properties", "template", "parameters", "ftpsState", "defaultValue" }, DisplayName = "One level nested template parameter found")]
+        [DataRow("resources[0].resourceGroup", 2, new object[] { "resources", 0, "properties", "template", "resources", 1, "properties", "template", "resources", 0, "resourceGroup" }, DisplayName = "Two levels nested template resource property found")]
+        [DataRow("resources[0].properties.expressionEvaluationOptions", 2, new object[] { "resources", 0, "properties", "template", "resources", 1, "properties", "template", "resources", 0, "properties", "expressionEvaluationOptions" }, DisplayName = "Two levels nested template resource property found")]
+        [DataRow("resources[1].properties", 2, new object[] { "resources", 0, "properties", "template", "resources", 1, "properties", "template", "resources", 0, "properties" }, DisplayName = "Two levels nested template resource property found")]
+        [DataRow("contentVersion", 3, new object[] { "resources", 0, "properties", "template", "resources", 1, "properties", "template", "resources", 0, "properties", "template", "contentVersion" }, DisplayName = "Three levels nested template property found")]
+        [DataRow("resources[0].location", 3, new object[] { "resources", 0, "properties", "template", "resources", 1, "properties", "template", "resources", 0, "properties", "template", "resources", 0, "location" }, DisplayName = "Three levels nested template resource property found")]
+        [DataRow("resources[2].properties", 3, new object[] { "resources", 0, "properties", "template", "resources", 1, "properties", "template", "resources", 0, "properties", "template", "resources", 0, "properties" }, DisplayName = "Original resource found")]
+        public void ResolveLineNumber_TemplatesWithNestedTemplates_ReturnsCorrectLineNumber(string pathInTheExpandedTemplate, int numberOfTemplate, object[] pathInTheOriginalTemplate)
```

Comment:
There's no testing for dependent resources copied into parents.

Since there are complex scenarios needing tests, it's probably best to utilize the real template processor instead of artifically creating an expanded template here.  We can always do something like save the expanded template to disk after generating to help with debugging test failures.  It will be much simpler to generate tests for complex scenarios if we don't have to worry about getting the test expanded template "just right".  It means we also don't have to manually create the resource mappings either.